### PR TITLE
CA-195429: logrotate xensource.log when over 100M

### PR DESCRIPTION
**Pull request by thomassa**
A new cron job calls a new shell-script which calls logrotate
with a new configuration file (based on the old one).

The job runs once per minute, and as well as rotating daily,
it rotates if the file is larger than 100M.

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
Looks like the right shape for what we need. What's the cost of running when the file is less than 100M? Do we need to run less than once per minute?

**Comment by thomassa** [comment]
I've not measured (yet) but I believe it's cheap. A simple shell-script starts logrotate which reads a simple config file, checks the size of xensource.log, sees that it is small enough, and exits.

The much more complex script that we used on some of the 6.x releases was also scheduled once per minute, and I seem to remember I did look at the performance of that.

**Comment by thomassa** [comment]
@xen-git retest this please

**Comment by thomassa** [comment]
Closing for now, until I've done some more dev-testing.

**Comment by thomassa** [comment]
Measurement of the cost of running the rotation cron-job when there is nothing it needs to do:
```
[root@xrtuk-09-02 log]# time /etc/xapi.d/xapi-logrotate.sh

real	0m0.016s
user	0m0.008s
sys	0m0.004s
```
```
[root@xrtuk-09-02 log]# time /etc/xapi.d/xapi-logrotate.sh

real	0m0.017s
user	0m0.000s
sys	0m0.012s
```

**Comment by thomassa** [comment]
(The commits will need squashing. Also @jonludlam was concerned that running once per minute might be too often even though normally it will take <20milliseconds and do nothing.)

**Comment by thomassa** [comment]
I've changed it to run once per hour.

**Comment by jonludlam** [comment]
I don't think xapi.d is the right place for a new script - @euanh is specfile tsar - libexec perhaps? Other than that it looks good.

File: xapi.spec.in [hunk 1]
```diff
@@ -222,6 +222,8 @@ rm -rf $RPM_BUILD_ROOT
 /etc/xapi.d/plugins/install-supp-pack
 /etc/xapi.d/plugins/disk-space
 /etc/xapi.d/extensions
+/etc/xapi.d/xapi-logrotate.conf
+%attr(0755,root,root) /etc/xapi.d/xapi-logrotate.sh
```
Comment (euanh) [inline xapi.spec.in]: I think this should probably be %{_libexecdir}/%{name}/xapi-logrotate.sh, as described 
[in the packaging guidelines.](https://fedoraproject.org/wiki/Packaging:Guidelines#Libexecdir)
You may find that you don't need the %attr specification if you put it there, as long as it has the right attributes when installed by the build script.

**Comment by thomassa** [comment]
@xen-git retest this please

**Comment by thomassa** [comment]
@xen-git retest this please

**Comment by thomassa** [comment]
I've updated the places for `xapi-logrotate.sh` and also `xapi-logrotate.conf`.
Not yet tested that the resulting RPM works as expected, but have I made decent choices for the paths now?

**Comment by thomassa** [comment]
@xen-git retest this please

**Comment by thomassa** [comment]
Dev-testing redone with the latest changes. After upgrading the xapi-core rpm, there is a file at `/etc/cron.d/xapi-logrotate.cron` with contents
```
# Run at 40 minutes past the hour,
# every hour, day-of-month, month, day-of-week
40   *  *  *  * root /opt/xensource/libexec/xapi-logrotate.sh
```
I have not waited for cron to schedule the job, but running `/opt/xensource/libexec/xapi-logrotate.sh` does the right thing; the file has the intended permission flags.
There is no `xapi` under `/etc/logrotate.d`.

**Comment by euanh** [comment]
:shipit:

**Comment by simonjbeaumont** [comment]
when the :snowflake: has thawed of course.

**Pull request merged**